zerver/lib/actions.py

```diff
@@ -3642,6 +3642,8 @@ def do_remove_realm_alias(alias):
     realm = alias.realm
     domain = alias.domain
     alias.delete()
+    if RealmAlias.objects.filter(realm=realm).count() == 0:
+        do_set_realm_restricted_to_domain(realm, False)
```

Comment:
It will be needed to update the state on the frontend. Since we are setting `restricted_to_domain` to false automatically in case there are no more aliases, we should update the admin about it.
I will comment it to make it more clear.